message.go

```diff
@@ -0,0 +1,63 @@
+package legs
+
+import (
+	"bytes"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-varint"
+)
+
+func encodeMessage(c cid.Cid, addrs []multiaddr.Multiaddr) []byte {
```

Comment:
would it make sense to define a schema and cbor encode the data, rather than doing it by hand?

Reply:
cbor_gen does not support slices, which I need to encode `addrs`, so I would need to write my own cbor encoding.  I was also worried about keeping the message as small as possible for getting spread via gossip. I did define a message datatype.  LMK if you would still prefer a cbor or other type of encoder.

Reply:
not blocking, but we may eventually want to re-visit